dev/com.ibm.ws.opentracing/src/com/ibm/ws/opentracing/OpentracingUserFeatureAccessService.java

```diff
@@ -56,24 +69,39 @@ public void setOpentracingTracerFactory(OpentracingTracerFactory opentracingTrac
      */
     @Trivial
     public static Tracer getTracerInstance(String appName) {
-        String prefix = "getTracerInstance";
-        if ( TraceComponent.isAnyTracingEnabled() && tc.isEntryEnabled() ) {
-            Tr.entry(tc, prefix, appName);
+        String methodName = "getTracerInstance";
+        if (TraceComponent.isAnyTracingEnabled() && tc.isEntryEnabled()) {
+            Tr.entry(tc, methodName, appName);
         }
 
-        Tracer tracer;
-        try {
-            tracer = opentracingTracerFactory.newInstance(appName);
-            if ( tracer == null ) {
-                Tr.error(tc, "OPENTRACING_TRACERFACTORY_RETURNED_NULL");
+        Tracer tracer = null;
+
+        if (opentracingTracerFactory != null) {
+            try {
+                tracer = opentracingTracerFactory.newInstance(appName);
+            } catch (Throwable t) {
+                Tr.error(tc, "OPENTRACING_COULD_NOT_CREATE_TRACER", t);
+            }
+        }
+
+        // It's not worth synchronizing around this and creating a
+        // bottleneck, since the only purpose is to print an error
+        // message if there is no factory configured. In the worst
+        // (and unlikely) case, this may cause a handful of those error messages to
+        // be printed instead of the expected 1, and this is fine
+        // to avoid the performance impact.
+        if (factoryFirstUse) {
+            factoryFirstUse = false;
+
+            if (opentracingTracerFactory == null) {
+                if (TraceComponent.isAnyTracingEnabled() && tc.isErrorEnabled()) {
```

Comment:
It is wrong to require trace enablement in order to log an error message.  Also, Tr.error is considered not to be a performance path, and so the tc.isErrorEnabled() is unnecessary as well, as Tr.error will check for that.